### PR TITLE
Search by state

### PR DESCRIPTION
**Pull request by analeamaesterling**

### CODE REVIEW

File: src/main/ui/src/App.js [hunk 1]
```diff
@@ -1,3 +1,16 @@
+import SearchByState from './components/SearchByState';
```
Comment (jaylingelbach) [inline src/main/ui/src/App.js:1]: This doesn't look right. When I open up vscode it says 
`Parsing error: Identifier 'App' has already been declared. (20:6)`

File: src/main/ui/src/components/SearchByState.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    .catch((error) => console.log(error));
+}, [searchTerm]);
+
+    function handleSearch(e) {
+        e.preventDefault();
+        setSearchTerm(e.target.value);
+    }
+
+    if (data.length > 0) {
+        let url = data.filter((i) => {
```
Comment (jaylingelbach) [inline src/main/ui/src/components/SearchByState.jsx:23]: VSCode error `'url' is declared but its value is never read.`

File: src/main/ui/src/index.js [hunk 3]
```diff
@@ -1,4 +1,11 @@
 
+import React from 'react';
```
Comment (jaylingelbach) [inline src/main/ui/src/index.js:2]: A bunch wrong from the merge conflict here.
Screen Shot 2022-09-01 at 10 02 40 PM

**Review by jaylingelbach — CHANGES_REQUESTED** [review]
After doing a merge conflict, you should try to run the program to make sure things didn't break :)

File: src/main/ui/src/index.js (hunk 3 above)
Comment (Dbeaven405) [reply src/main/ui/src/index.js]: fixed

File: src/main/ui/src/components/SearchByState.jsx (hunk 2 above)
Comment (Dbeaven405) [reply src/main/ui/src/components/SearchByState.jsx]: url removed

File: src/main/ui/src/App.js (hunk 1 above)
Comment (Dbeaven405) [reply src/main/ui/src/App.js]: this is what was causing the search to be repeated.. it was doubled in index.js

File: src/main/ui/src/components/SearchByState/SearchByState.jsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        if (searchTerm === "") {
+            return park
+        } if (park.states.toLowerCase().includes(searchTerm.toLowerCase())) {
+            return park
+        }
+    }
+
+return (
+<div>
+    <input
```
Comment (jaylingelbach) [inline src/main/ui/src/components/SearchByState/SearchByState.jsx:38]: In the requirements in the story and in discussion we had talked about making this a dropdown. Why did we not do that?

**Review by jaylingelbach — CHANGES_REQUESTED** [review]
Change to dropdown of states as discussed.

File: src/main/ui/src/components/SearchByState/SearchByState.jsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+     fetch(url)
+         .then((response) => response.json())
+         .then((json) => {
+             setData(json.data);
+         })
+     .catch((error) => console.log(error));
+ }, [selectState]);
+
+     if (data.length > 0) {
+         data.filter((i) => {
```
Comment (jaylingelbach) [inline src/main/ui/src/components/SearchByState/SearchByState.jsx:19]: please use a more descriptive variable name `i` doesn't really let me know what you are iterating through.

**Review by jaylingelbach — CHANGES_REQUESTED** [review]
Just one little change and this can be merged. Make sure you mark all of the issues as resolved so I can merge it when finised.

**Comment by jaylingelbach** [comment]
Screen Shot 2022-09-05 at 4 12 23 PM

5 unresolved conversations.

**Review by jaylingelbach — APPROVED** [review]

File: src/main/ui/src/App.css [hunk 6]
```diff
@@ -3,6 +3,8 @@
 }
 
 
+
+
```
Comment (jaylingelbach) [inline src/main/ui/src/App.css:7]: remove plz

File: src/main/ui/src/components/SearchByState/SearchByState.jsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
+import React from 'react';
```
Comment (jaylingelbach) [inline src/main/ui/src/components/SearchByState/SearchByState.jsx:1]: I noticed that on the app startup that the state alabama is selected in the dropdown, but all states show in the list. Please change the initial selection to `Choose a state` that way all the states showing makes sense. Also I see ZERO styling on this. Let's look at emily/candide's work and match it up please.

Comment (jaylingelbach) [reply src/main/ui/src/components/SearchByState/SearchByState.jsx:1]: Also make sure to remove all comments before submitting

Thank you!

**Pull request merged**